Slim down D2k weapon yamls

reaperrr opened the pull request:
This downsizes the weapon yaml files and makes it a bit easier to apply wider-scale changes like #12513 later.

With regard to missile commit: `^Rocket` = small + unguided, `^Missile` = larger + guided.

reaperrr committed:
Slim down D2k Debris weapons via inheritance

reaperrr committed:
Slim down D2k cannon weapons via inheritance

reaperrr committed:
Slim down D2k miniguns via inheritance

reaperrr committed:
Slim down D2k missile weapons via inheritance

reaperrr committed:
Slim down D2k CrateNuke via inheritance

pchote commented on mods/d2k/weapons/largeguns.yaml:
```diff
@@ -57,41 +26,57 @@
 	ReloadDelay: 45
 
 DevBullet:
+	Inherits: ^Cannon
```
This changes `Inaccuracy` from undefined (i.e. 0) to 380.

pchote commented on mods/d2k/weapons/missiles.yaml:
```diff
@@ -32,147 +33,105 @@ Bazooka:
+		Explosions: small_explosion
+
+Bazooka:
+	Inherits: ^Rocket
```
This changes `HorizontalRateOfTurn` from 1 to 0.

reaperrr replied:
They weren't tracking in the original, this is intentional.

pchote replied:
Ah, I see.  Would it make sense to change these to an arcing Bullet like we use for non-guided projectiles in RA and TD?

reaperrr replied:
I guess it wouldn't hurt.

pchote commented on mods/d2k/weapons/missiles.yaml:
```diff
@@ -32,147 +33,105 @@ Bazooka:
-		ImpactSounds: EXPLSML1.WAV
 
 TowerMissile:
+	Inherits: ^Missile
```
This changes `TrailPalette` from undefined to `effect75alpha`.  Probably a good thing (I haven't tested)?

reaperrr replied:
Intentional, yes. They were transparent in the original.

pchote commented on mods/d2k/weapons/missiles.yaml:
```diff
@@ -32,147 +33,105 @@ Bazooka:
-		CruiseAltitude: 1c0
-		MinimumLaunchAngle: 64
-		VerticalRateOfTurn: 10
-		Inaccuracy: 96
```
This changes the inaccuracy to 384.

reaperrr replied:
Fixed.

pchote commented on mods/d2k/weapons/missiles.yaml:
```diff
@@ -32,147 +33,105 @@ Bazooka:
-		ImpactSounds: EXPLSML1.WAV
 
 mtank_pri:
+	Inherits: ^Missile
```
This changes `TrailPalette` from undefined to `effect75alpha`.

reaperrr replied:
See above.

pchote commented on mods/d2k/weapons/missiles.yaml:
```diff
@@ -32,147 +33,105 @@ Bazooka:
-		ImpactSounds: EXPLSML1.WAV
 
 DeviatorMissile:
+	Inherits: ^Missile
```
This adds a `MinRange: 1c0` that didn't exist before.  Probably a good thing.

pchote commented on mods/d2k/weapons/smallguns.yaml:
```diff
@@ -21,143 +21,61 @@ LMG:
 
 Fremen_L:
-	ReloadDelay: 40
-	Delay: 5
```
This removes the delay completely (it isn't defined in `M_HMG`).

reaperrr replied:
For weapons, just `Delay` is a bogus property. I haven't checked where this came from, but in all likelyness a rebase error.

pchote commented on mods/d2k/weapons/smallguns.yaml:
```diff
@@ -21,143 +21,61 @@ LMG:
-	Warhead@2Eff: CreateEffect
-		Explosions: piffs
 
 HMG_muzzle:
```
My comparison shows that these are missing the `InstantHit` projectile, so I guess you need to rebase.

reaperrr replied:
Ah, right, forgot about that.

pchote reviewed (changes requested):
Some minor regressions to fix, otherwise looks good.

reaperrr committed:
Remove explosion sound from D2k gun turret weapon

It was soundless in the original, and the sound is a bit annoying, too.

reaperrr commented:
Rebased and updated.

Made some changes:
- unguided rockets now use Bullet
- reduced missile `MinRange` to `0c512`, otherwise they'd miss diagonally passing vehicles and adjacent infantry on closer subcells.
- Removed explosion sound from gun turret cannon, because the sound was annoying and the original didn't have it, either.

abcdefg30 commented:
Looks good to me. :+1: / :white_check_mark:
